pkg/serverless/metrics/metric_test.go

```diff
@@ -21,17 +22,24 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/aggregator"
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/dogstatsd"
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
 )
 
+func TestMain(m *testing.M) {
+	// setting the hostname cache saves about 1s when starting the metric agent
+	cacheKey := cache.BuildAgentKey("hostname")
+	cache.Cache.Set(cacheKey, hostname.Data{}, cache.NoExpiration)
```

Comment:
At least when I run the tests locally, I usually will not include our serverless build tag. This means that normal hostname resolution will occur. This prevents that from happening.